rcl/src/rcl/publisher.c

```diff
@@ -137,6 +140,19 @@ rcl_publisher_init(
     (const void *)publisher->impl->rmw_handle,
     remapped_topic_name,
     options->qos.depth);
+  // Register type.
+  if (rcl_node_type_cache_is_valid(node)) {
```

Comment:
If I recall our conversation from earlier this week, I think we determined that we are just going to unconditionally register types.  If that recollection is correct, then we can just remove this conditional.